contracts/paymaster/DeployPaymaster.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "../factory/ProxyFactory.sol";
+import "./BasePaymaster.sol";
+
+/**
+ * A paymaster to be used on deploys.
+ * - We maintain the interface but not the functions, mainly to be
+ * - GSN compatible
+ */
+contract DeployPaymaster is BasePaymaster {
+
+    address private factory;
+
+    constructor(address proxyFactory) public {
+        factory = proxyFactory;
+    }
+
+    function versionPaymaster() external override virtual view returns (string memory){
+        return "2.0.1+opengsn.token.ipaymaster";
+    }
+
+    mapping (address => bool) public tokens;
+
+    function preRelayedCall(
+        GsnTypes.RelayRequest calldata relayRequest,
+        bytes calldata signature,
+        bytes calldata approvalData,
+        uint256 maxPossibleGas
+    )
+    external 
+    override 
+    virtual
+    relayHubOnly
+    returns (bytes memory context, bool revertOnRecipientRevert) {
+        require(tokens[relayRequest.request.tokenContract], "Token contract not allowed");
+        IERC20 token = IERC20(relayRequest.request.tokenContract);
+
+        require(address(relayRequest.request.factory) != address(0), "factory should be defined");
+        require(address(relayRequest.request.factory) == factory, "factory should be the trusted one!");
+        
+        address contractAddr = ProxyFactory(relayRequest.request.factory).getSmartWalletAddress(relayRequest.request.from, relayRequest.request.to, relayRequest.request.data);
```

Comment:
a new PR to master added new parameters, "Index" and "recoverer" to getSmartWalletAddress

Reply:
As soon as it gets approved, your PR wont merge

Reply:
https://github.com/rsksmart/enveloping/pull/33

Reply:
fixed